### PR TITLE
chore: use pgx v5

### PR DESCRIPTION
**Pull request by adlerhurst**
Upgrades the major version of the database driver

closes https://github.com/zitadel/zitadel/issues/7254

### Definition of Ready

- [x] I am happy with the code
- [x] Short description of the feature/issue is added in the pr description
- [x] PR is linked to the corresponding user story
- [x] Acceptance criteria are met
- [x] All open todos and follow ups are defined in a new ticket and justified
- [x] Deviations from the acceptance criteria and design are agreed with the PO and documented.
- [x] No debug or dead code
- [x] My code has no repetitions
- [x] Critical parts are tested automatically
- [x] Where possible E2E tests are implemented
- [x] Documentation/examples are up-to-date
- [x] All non-functional requirements are met
- [x] Functionality of the acceptance criteria is checked manually on the dev system.

### CODE REVIEW

**Review by muhlemmer — COMMENTED** [review]
Nice job on dealign with the array types! Couldn't figure this on out myself ;)

Just 2 questions.

File: go.mod [hunk 1]
```diff
@@ -3,7 +3,7 @@ module github.com/zitadel/zitadel
 go 1.21.1
 
 // https://go.dev/doc/toolchain
-toolchain go1.21.8
+toolchain go1.21.5
```
[the remark this answers is not in the document]
Comment (adlerhurst) [reply go.mod]: 1.21.8 hopefully works :D

File: internal/database/type.go [hunk 2]
```diff
@@ -96,14 +135,35 @@ type Duration time.Duration
...
 		return err
 	}
-	*d = Duration(time.Duration(interval.Microseconds*1000) + time.Duration(interval.Days)*24*time.Hour + time.Duration(interval.Months)*30*24*time.Hour)
+	*d = intervalToDuration(interval)
 	return nil
 }
 
+func intervalToDuration(interval *pgtype.Interval) Duration {
+	return Duration(time.Duration(interval.Microseconds*1000) + time.Duration(interval.Days)*24*time.Hour + time.Duration(interval.Months)*30*24*time.Hour)
+}
```
[the remark this answers is not in the document]
Comment (adlerhurst) [reply internal/database/type.go:165]: the problem is that our tests use different types. Pgx itself always passes a int64. Thats why the type switch is used at all. I think there is one test which passes interval

[the remark this answers is not in the document]
Comment (adlerhurst) [reply internal/database/type.go:165]: what we could try is to extend `mock.ConvertValue`-function that it converts all the wrong types to int64

[the remark this answers is not in the document]
Comment (muhlemmer) [reply internal/database/type.go:165]: If it's just for testing, I'm fine with it.

**Review by muhlemmer — APPROVED** [review]

**Review by muhlemmer — APPROVED** [review]

**Pull request merged**